Document the connection parameters.

knz opened the pull request:
Informs  #283.
Informs #631.
Informs  #2458.
Informs  #2751.
Fixes #2781.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/docs/2853)
<!-- Reviewable:end -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/dcf467a6dfe6277d5faf233e973f2782d79d90ae/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/82e8a1971a69503995224ee29ce1031fd177bbf6/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/a3258715fb81831f698ffd8afef744f28fb53135/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/46181c0ad8e9a7d163dcf796fb36a29a80f5dd39/

jseldess commented on v1.1/configure-replication-zones.md:
```diff
@@ -135,16 +135,10 @@ The `zone` command and subcommands support the following [general-use](#general)
-`--port`<br>`-p` | The server port to connect to. <br><br>**Env Variable:** `COCKROACH_PORT`<br>**Default:** `26257`
-`--url` | The connection URL. If you use this flag, do not set any other connection flags.<br><br>For insecure connections, the URL format is: <br>`--url=postgresql://<user>@<host>:<port>/<database>?sslmode=disable`<br><br>For secure connections, the URL format is:<br>`--url=postgresql://<user>@<host>:<port>/<database>`<br>with the following parameters in the query string:<br>`sslcert=<path-to-cl ...
-`--user`<br>`-u` | The user connecting to the database. Currently, only the `root` user can configure replication zones. <br><br>**Env Variable:** `COCKROACH_USER`<br>**Default:** `root`
+`--url`, `--host`, `--port`, `--user`, `--database`, `--certs-dir`, `--insecure`,  | See [Client Connection Parameters](connection-parameters.html) for details. Currently, only the `root` user can configure replication zones and the `--database` flag is not effective.
```
I have some concerns about this approach:

1. On the one hand, I don't like bouncing users to a separate page to learn about the flags for a specific `cockroach` command. That's a burden that a reference doc should try to avoid. As an alternative, I'd prefer to leave the flags on the command pages, grouped under a new `Client Connection` header, and have a note at the top of that section calling out the option to use a url instead with a link to the new page. I'd like to hear your response to this suggestion before reviewing much more. Would also be interested in @bdarnell or @mberhault's opinion here.

2. But if we have to do this, I'd suggest we change the presentation to either:
    1. Use `<br>` to put each flag on its own line in the cell.
    2. Add a distinct section called **Client Connection** and add a sentence with a link to the page you created.

jseldess commented on v1.1/connection-parameters.md:
```diff
@@ -0,0 +1,198 @@
+- Most client apps, including most `cockroach` commands, use a SQL connection
+  established via a [PostgreSQL connection URL](#connect-using-a-url). When using a URL,
+  a client can also specify SSL/TLS settings and additional SQL-level parameters. This mode provides the most configuration flexibility.
+- Most `cockroach` commands also provide [discrete connection parameters](#connect-using-discrete-parameters) that
```
Instead of `discrete connection parameters`, I'd prefer we be direct and say `command-line flags`? 

If you're on board, please change it throughout this page.

jseldess commented on v1.1/connection-parameters.md:
```diff
@@ -0,0 +1,198 @@
+
+Client | Supports [connection by URL](#connect-using-a-url) | Supports [discrete connection parameters](#connect-using-discrete-parameters)
+-------|----------------------------|-----------------------------------
+Client apps using a PostgreSQL driver | ✓ | (✗)
```
Any reason the `x` is in parentheses?

jseldess commented on v1.1/connection-parameters.md:
```diff
@@ -0,0 +1,198 @@
+postgres://<username>:<password>@<host>:<port>/<database>?<parameters>
+~~~
+
+Component | Description
```
Instead of putting `optional` in the description, I'd suggest a third column, `Required`, using the checks and xs to indicate whether required or not.

a-robinson commented:
I don't have much to add beyond @jseldess's comments

---

Reviewed 29 of 29 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

*[v1.1/configure-replication-zones.md, line 141 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9FzERath47ruOI6HyO-r1-141:-L9H7TZ-Dkyfkw6F-j9B:b-ocqhvi) ([raw file](https://github.com/cockroachdb/docs/blob/46181c0ad8e9a7d163dcf796fb36a29a80f5dd39/v1.1/configure-replication-zones.md#L141)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

I have some concerns about this approach:

1. On the one hand, I don't like bouncing users to a separate page to learn about the flags for a specific `cockroach` command. That's a burden that a reference doc should try to avoid. As an alternative, I'd prefer to leave the flags on the command pages, grouped under a new `Client Connection` header, and have a note at the top of that section calling out the option to use a url instead with a link to the new page. I'd like to hear your response to this suggestion before reviewing much more. Would also be interested in @bdarnell or @mberhault's opinion here.

2. But if we have to do this, I'd suggest we change the presentation to either:
    1. Use `<br>` to put each flag on its own line in the cell.
    2. Add a distinct section called **Client Connection** and add a sentence with a link to the page you created.
</blockquote></details>

+1 for Jesse's suggested approach. Continue defining the connection flags here, with a note that `--url` can be set instead, linking to the connection parameters page as an in-depth reference for it.

---

*[v1.1/connection-parameters.md, line 79 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9HBwPa4WprflsGDA0H:-L9HBwPa4WprflsGDA0I:b-rjnxj6) ([raw file](https://github.com/cockroachdb/docs/blob/46181c0ad8e9a7d163dcf796fb36a29a80f5dd39/v1.1/connection-parameters.md#L79)):*
> ```Markdown
> `application_name` | An initial value for the [`application_name` session variable](set-vars.html). | Empty string.
> `sslmode` | Which type of secure connection to use: `disable`, `allow`, `prefer`, `require`, `verify-ca` or `verify-full`. See below for details. | `disable`
> `sslrootcert` | Client-side path to the CA certificate, when `sslmode` is not `disable`. | Empty string.
> ```

Are we ok assuming that people will know what the "CA certificate", "client certificate", and "client private key" are without introducing them? We may want to squeeze a reference to create-security-certificates.html in somewhere as background reading.

---

*[v1.1/connection-parameters.md, line 153 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9HCh0n4F_aR6K4tAUr:-L9HCh0oCSXfDNkrMv1W:b-fggrvc) ([raw file](https://github.com/cockroachdb/docs/blob/46181c0ad8e9a7d163dcf796fb36a29a80f5dd39/v1.1/connection-parameters.md#L153)):*
> ```Markdown
> 
> ~~~
> --user root \
> ```

nit, but why this indentation difference here (and in the secure example below)?

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-:-L9HDDcX1KgxElZeNv1B:b5ncfhn)*
<!-- Sent from Reviewable.io -->

knz commented:
Restructured to include the parameters on every client command page via an include. PTAL.

---

Review status: 13 of 31 files reviewed at latest revision, 6 unresolved discussions.

---

*[v1.1/configure-replication-zones.md, line 141 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9FzERath47ruOI6HyO-r1-141:-L9HMwGj8s88rtrAmgcI:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/46181c0ad8e9a7d163dcf796fb36a29a80f5dd39/v1.1/configure-replication-zones.md#L141)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

+1 for Jesse's suggested approach. Continue defining the connection flags here, with a note that `--url` can be set instead, linking to the connection parameters page as an in-depth reference for it.
</blockquote></details>

Done.

---

*[v1.1/connection-parameters.md, line 23 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9FzERath47ruOI6HyP-r1-23:-L9HNqhz1VxQvB6WJMXI:b-kvmvw6) ([raw file](https://github.com/cockroachdb/docs/blob/46181c0ad8e9a7d163dcf796fb36a29a80f5dd39/v1.1/connection-parameters.md#L23)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Instead of `discrete connection parameters`, I'd prefer we be direct and say `command-line flags`? 

If you're on board, please change it throughout this page.
</blockquote></details>

But the thing is, `--url` is also a command-line flag. The contrast is between the "combined" nature of the `--url` flag and the "discrete" / "separate" nature of the other flags.

---

*[v1.1/connection-parameters.md, line 33 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9FzERath47ruOI6HyP-r1-33:-L9HNZSS1iaA2s3dYGuz:b-eowr3y) ([raw file](https://github.com/cockroachdb/docs/blob/46181c0ad8e9a7d163dcf796fb36a29a80f5dd39/v1.1/connection-parameters.md#L33)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Any reason the `x` is in parentheses?
</blockquote></details>

Clarified.

---

*[v1.1/connection-parameters.md, line 53 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9FzERath47ruOI6HyP-r1-53:-L9HN_dr48CuMHdfhaMO:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/46181c0ad8e9a7d163dcf796fb36a29a80f5dd39/v1.1/connection-parameters.md#L53)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Instead of putting `optional` in the description, I'd suggest a third column, `Required`, using the checks and xs to indicate whether required or not.
</blockquote></details>

Done.

---

*[v1.1/connection-parameters.md, line 79 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9HBwPa4WprflsGDA0H:-L9HNlmWA4kygODxyGKc:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/46181c0ad8e9a7d163dcf796fb36a29a80f5dd39/v1.1/connection-parameters.md#L79)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

Are we ok assuming that people will know what the "CA certificate", "client certificate", and "client private key" are without introducing them? We may want to squeeze a reference to create-security-certificates.html in somewhere as background reading.
</blockquote></details>

Done.

---

*[v1.1/connection-parameters.md, line 153 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9HCh0n4F_aR6K4tAUr:-L9HNmTj1ppo4oSI4teS:by6idoo) ([raw file](https://github.com/cockroachdb/docs/blob/46181c0ad8e9a7d163dcf796fb36a29a80f5dd39/v1.1/connection-parameters.md#L153)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

nit, but why this indentation difference here (and in the secure example below)?
</blockquote></details>

To make it clear the lines below are a continuation?

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-:-L9HO0lK-0mBAtoziIPW:b-dmvvfq)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/9fad252f4573236d3469f4ed9b6896c5b2fa50d7/

jseldess commented:
Review status: 13 of 31 files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[_includes/sql/v2.0/connection-parameters.md, line 5 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9K4tp0Aqflytvmb--Y:-L9K4tp0Aqflytvmb--Z:b-rgnmv) ([raw file](https://github.com/cockroachdb/docs/blob/9fad252f4573236d3469f4ed9b6896c5b2fa50d7/_includes/sql/v2.0/connection-parameters.md#L5)):*
> ```Markdown
> `--host` | The server host to connect to. This can be the address of any node in the cluster. <br><br>**Env Variable:** `COCKROACH_HOST`<br>**Default:**`localhost`
> `--port`<br>`-p` | The server port to connect to. <br><br>**Env Variable:** `COCKROACH_PORT`<br>**Default:** `26257`
> `--user`<br>`-u` | The [SQL user](create-and-manage-users.html) that will own the client session.<br><br>**2Env Variable:** `COCKROACH_USER`<br>**Default:** `root`
> ```

`2Env Variable` > `Env Variable`

---

*[_includes/sql/v2.0/connection-parameters.md, line 6 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9K4ycE9_y7M0FmUUiK:-L9K4ycE9_y7M0FmUUiL:bs32une) ([raw file](https://github.com/cockroachdb/docs/blob/9fad252f4573236d3469f4ed9b6896c5b2fa50d7/_includes/sql/v2.0/connection-parameters.md#L6)):*
> ```Markdown
> `--port`<br>`-p` | The server port to connect to. <br><br>**Env Variable:** `COCKROACH_PORT`<br>**Default:** `26257`
> `--user`<br>`-u` | The [SQL user](create-and-manage-users.html) that will own the client session.<br><br>**2Env Variable:** `COCKROACH_USER`<br>**Default:** `root`
> `--password` | Enable password authentication for the user; you will be prompted to enter the password on the command line.<br/><br/><span class="version-tag">Changed in v2.0:</span> Password creation is supported only in secure clusters for non-`root` users. The `root` user must authenticate with a client certificate and key.<br/><br/>[Find more detail about how CockroachDB handles passwords](create-and-manage-users.html#user-authentication).
> ```

This flag isn't available for most client commands. To my knowledge, it's only valid for `cockroach user` and `cockroach sql`. So we probably need to clarify that somehow. 

The same is probably true of some of the others. I'd need to audit them all to figure that out, though.

---

*[v1.1/connection-parameters.md, line 23 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9FzERath47ruOI6HyP-r1-23:-L9K3SG44hX6Jxmn5kkz:b-w9a10) ([raw file](https://github.com/cockroachdb/docs/blob/46181c0ad8e9a7d163dcf796fb36a29a80f5dd39/v1.1/connection-parameters.md#L23)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

But the thing is, `--url` is also a command-line flag. The contrast is between the "combined" nature of the `--url` flag and the "discrete" / "separate" nature of the other flags.
</blockquote></details>

Ah, ok. The current language makes sense then. Thanks for the clarification, @knz.

---

*[v1.1/connection-parameters.md, line 33 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9FzERath47ruOI6HyP-r1-33:-L9K3Wo5AcwWF7giikIm:b5mvb03) ([raw file](https://github.com/cockroachdb/docs/blob/46181c0ad8e9a7d163dcf796fb36a29a80f5dd39/v1.1/connection-parameters.md#L33)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Clarified.
</blockquote></details>

I don't see a clarification?

---

*[v1.1/connection-parameters.md, line 53 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9FzERath47ruOI6HyP-r1-53:-L9K41Sa-V1iXShqYTCk:bheiufh) ([raw file](https://github.com/cockroachdb/docs/blob/46181c0ad8e9a7d163dcf796fb36a29a80f5dd39/v1.1/connection-parameters.md#L53)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

I don't see this change yet.

---

*[v1.1/connection-parameters.md, line 79 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9HBwPa4WprflsGDA0H:-L9K4RNME71JoyK0hFTj:behgwyf) ([raw file](https://github.com/cockroachdb/docs/blob/46181c0ad8e9a7d163dcf796fb36a29a80f5dd39/v1.1/connection-parameters.md#L79)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

I don't see a change here?

---

*[v1.1/connection-parameters.md, line 29 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9K3n6JBGOyfMmxMy2O:-L9K3n6JBGOyfMmxMy2P:b-v5pbsy) ([raw file](https://github.com/cockroachdb/docs/blob/9fad252f4573236d3469f4ed9b6896c5b2fa50d7/v1.1/connection-parameters.md#L29)):*
> ```Markdown
>   discrete connection parameters.
> 
> The following table summarizes which command supports which connection parameters:
> ```

`which command` > `which client`?

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2853)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/c1a731ebbf57b0b8ece270c2147dc2d25c16cbf1/

knz commented:
RFAL

---

Review status: 13 of 31 files reviewed at latest revision, 9 unresolved discussions, some commit checks pending.

---

*[_includes/sql/v2.0/connection-parameters.md, line 5 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9K4tp0Aqflytvmb--Y:-L9L9FMq43BkbZSx9z6d:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/9fad252f4573236d3469f4ed9b6896c5b2fa50d7/_includes/sql/v2.0/connection-parameters.md#L5)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

`2Env Variable` > `Env Variable`
</blockquote></details>

Done.

---

*[_includes/sql/v2.0/connection-parameters.md, line 6 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9K4ycE9_y7M0FmUUiK:-L9L9OCR66sAZyLAPvHT:b-ffo8t4) ([raw file](https://github.com/cockroachdb/docs/blob/9fad252f4573236d3469f4ed9b6896c5b2fa50d7/_includes/sql/v2.0/connection-parameters.md#L6)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

This flag isn't available for most client commands. To my knowledge, it's only valid for `cockroach user` and `cockroach sql`. So we probably need to clarify that somehow. 

The same is probably true of some of the others. I'd need to audit them all to figure that out, though.
</blockquote></details>

d'oh. I checked the source code and it is indeed only valid for `user set`. Reverted.

---

*[v1.1/connection-parameters.md, line 33 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9FzERath47ruOI6HyP-r1-33:-L9LAfF53ZuorscdgUEY:budhgwg) ([raw file](https://github.com/cockroachdb/docs/blob/46181c0ad8e9a7d163dcf796fb36a29a80f5dd39/v1.1/connection-parameters.md#L33)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

I don't see a clarification?
</blockquote></details>

I mistakenly only changed the text in the 2.0 version, this is the 1.1 version. Modified both. 

---

*[v1.1/connection-parameters.md, line 53 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9FzERath47ruOI6HyP-r1-53:-L9LAkRu4oae5VsK-oK8:b2nreh6) ([raw file](https://github.com/cockroachdb/docs/blob/46181c0ad8e9a7d163dcf796fb36a29a80f5dd39/v1.1/connection-parameters.md#L53)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

I don't see this change yet.
</blockquote></details>

Same as above.

---

*[v1.1/connection-parameters.md, line 79 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9HBwPa4WprflsGDA0H:-L9LAmwPCQUkc05VjygJ:b2nreh6) ([raw file](https://github.com/cockroachdb/docs/blob/46181c0ad8e9a7d163dcf796fb36a29a80f5dd39/v1.1/connection-parameters.md#L79)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

I don't see a change here?
</blockquote></details>

Same as above.

---

*[v1.1/connection-parameters.md, line 29 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9K3n6JBGOyfMmxMy2O:-L9LAeg1E9acO1xnvwTP:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/9fad252f4573236d3469f4ed9b6896c5b2fa50d7/v1.1/connection-parameters.md#L29)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

`which command` > `which client`?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-:-L9LBO0e81vdWcnk-30f:bb72pd0)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/7a4a62b5ac8bddcd6a07216b0863b4dc9aa708ed/

knz commented:
Rebased to avoid conflicts.

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/c0a823b0d60700f1a043d929b35b229a359911ac/

jseldess commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> with some final nits and questions. Thanks for this work, @knz!

---

Review status: 12 of 31 files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

*[_includes/sql/v1.1/connection-parameters.md, line 6 at r3](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9PUC0FEjqK8h8uG5kQ:-L9PUC0FEjqK8h8uG5kR:btagcuq) ([raw file](https://github.com/cockroachdb/docs/blob/c0a823b0d60700f1a043d929b35b229a359911ac/_includes/sql/v1.1/connection-parameters.md#L6)):*
> ```Markdown
> `--port`<br>`-p` | The server port to connect to. <br><br>**Env Variable:** `COCKROACH_PORT`<br>**Default:** `26257`
> `--user`<br>`-u` | The [SQL user](create-and-manage-users.html) that will own the client session.<br><br>**Env Variable:** `COCKROACH_USER`<br>**Default:** `root`
> `--database`<br>`-d` | A database name to use as current database in the SQL session.<br><br>**Env Variable:** `COCKROACH_DATABASE`<br>**Default:** empty string.
> ```

I don't think this flag is supported in all client commands. I vaguely remember that there are some commands that list the flag but don't support it. Can you double-check?

---

*[v1.1/connection-parameters.md, line 71 at r3](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9PTLsvAidrVFxGTZmr:-L9PTLsvAidrVFxGTZms:b-s7wrmq) ([raw file](https://github.com/cockroachdb/docs/blob/c0a823b0d60700f1a043d929b35b229a359911ac/v1.1/connection-parameters.md#L71)):*
> ```Markdown
> as described below.{{site.data.alerts.end}}
> 
> ### Additional connection parameters
> ```

nit: Use title case: `Additional Connection Parameters`

---

*[v1.1/connection-parameters.md, line 78 at r3](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9PSaBj-bffJAbIpMzM:-L9PSaBj-bffJAbIpMzN:b-qtywvz) ([raw file](https://github.com/cockroachdb/docs/blob/c0a823b0d60700f1a043d929b35b229a359911ac/v1.1/connection-parameters.md#L78)):*
> ```Markdown
> ----------|-------------|---------------
> `application_name` | An initial value for the [`application_name` session variable](set-vars.html). | Empty string.
> `sslmode` | Which type of secure connection to use: `disable`, `allow`, `prefer`, `require`, `verify-ca` or `verify-full`. See below for details. | `disable`
> ```

nit: Instead of "See below for details", we could use a cross-reference.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-:-L9PUMRBBi0qik2di0MS:b4emqom)*
<!-- Sent from Reviewable.io -->

knz commented:
Review status: 12 of 31 files reviewed at latest revision, 7 unresolved discussions, all commit checks successful.

---

*[_includes/sql/v1.1/connection-parameters.md, line 6 at r3](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9PUC0FEjqK8h8uG5kQ:-L9QbtpC3VpUOwWj70NR:b9hvzux) ([raw file](https://github.com/cockroachdb/docs/blob/c0a823b0d60700f1a043d929b35b229a359911ac/_includes/sql/v1.1/connection-parameters.md#L6)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

I don't think this flag is supported in all client commands. I vaguely remember that there are some commands that list the flag but don't support it. Can you double-check?
</blockquote></details>

Yes I checked the source code and again you are right. `--database` is actually only supported for `cockroach sql`, and none of the other commands. In a way the previous documentation that suggested it exists for other cockroach commands was misleading/incorrect.

I removed it from this table, and added it to the `cockroach sql` docs. Thanks for spotting it.

---

*[v1.1/connection-parameters.md, line 71 at r3](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9PTLsvAidrVFxGTZmr:-L9QcDbK3MNTaO1AsU7F:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c0a823b0d60700f1a043d929b35b229a359911ac/v1.1/connection-parameters.md#L71)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: Use title case: `Additional Connection Parameters`
</blockquote></details>

Done.

---

*[v1.1/connection-parameters.md, line 78 at r3](https://beta.reviewable.io/reviews/cockroachdb/docs/2853#-L9PSaBj-bffJAbIpMzM:-L9QcMWU3lWmjI_Ogq6W:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c0a823b0d60700f1a043d929b35b229a359911ac/v1.1/connection-parameters.md#L78)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: Instead of "See below for details", we could use a cross-reference.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2853)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/bf2edc2398796db40d6797616a4da891c7ee4f43/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/39f622e59674b7476773ec9e530858a3a9880927/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/4d7577a1ddc49fba5b14eee3c4d4a3ce27297e7a/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/9d53004197026eafe7b2ef9af363b25dffd6dcbc/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f81229f12a3b1bb038262c86bc368ac4cbea28b3/

mberhault commented:
Sorry, I just saw this. FYI: `sllmode` values of `allow` and `prefer` are not supported by Go's lib/pq, including our sql shell. See [lib/pq source](https://github.com/lib/pq/blob/master/ssl.go#L50). This is due to the lack of support for creating a connection in plaintext and switching to SSL on the fly.

knz commented:
Interesting. Will follow up.
